File: deob-ast/src/main/java/dev/openrs2/deob/ast/util/TypeUtils.java
package dev.openrs2.deob.ast.util;

import com.github.javaparser.resolution.types.ResolvedType;

public final class TypeUtils {
	public static boolean isString(ResolvedType type) {
		return type.isReferenceType() && type.asReferenceType().getQualifiedName().equals("java.lang.String");
	}

	private TypeUtils() {
		/* empty */
	}
}

File: deob-ast/src/main/java/org/openrs2/deob/ast/util/TypeUtils.kt
package org.openrs2.deob.ast.util

import com.github.javaparser.resolution.types.ResolvedType

public fun ResolvedType.isString(): Boolean {
    return isReferenceType && asReferenceType().qualifiedName == "java.lang.String"
}

public fun ResolvedType.isClass(): Boolean {
    return isReferenceType && asReferenceType().qualifiedName == "java.lang.Class"
}
